test/connectors/FinancialInstitutionsConnectorSpec.scala

```diff
@@ -73,14 +76,40 @@ class FinancialInstitutionsConnectorSpec extends SpecBase with WireMockServerHan
       result.futureValue.status mustBe OK
     }
 
-    "must return status as OK for addFi" in {
-      stubPostResponse(
-        s"/crs-fatca-fi-management/financial-institutions/create",
-        OK,
-        "{}"
-      )
-      val result = connector.addFi(createFIDetails)
-      result.futureValue.status mustBe OK
+    "addFi" - {
+
+      "must return Right(HttpResponse) when the response status is OK" in {
+        stubPostResponse(
+          s"/crs-fatca-fi-management/financial-institutions/create",
+          OK,
+          "{}"
+        )
+        val result = connector.addFi(createFIDetails).futureValue
+        result.equals(Future(Right(HttpResponse(OK))))
+      }
+
+      "return Left(ErrorDetails) when the response status is not OK" in {
+//todo: fix
+        val fiDetails = createFIDetails
+
+        val errorResponseJson =
+          """{
+            |"ErrorDetails": {
+            |    "timestamp": "2016-08-16T18:15:41Z",
+            |    "correlationId": "",
+            |    "errorCode": "503"
+            |}}""".stripMargin
+
+        stubPostResponse(
+          s"/crs-fatca-fi-management/financial-institutions/create",
+          SERVICE_UNAVAILABLE,
+          errorResponseJson
+        )
+        val result = connector.addFi(fiDetails).futureValue
+
+        result mustBe a[Left[_, _]]
+        result.left.get mustBe ErrorDetails("2016-08-16T18:15:41Z", "", Some("503"))
```

Comment:
the `connector.fi`, won't simply goto `Left` case but instead throwing an exception. In case if you want to return `Left`, you have to also add `.recoverWith` within the connector. Otherwise test it like this:

```
        connector.addFi(fiDetails).failed.map {
          case e: UpstreamErrorResponse =>
            e.statusCode mustBe SERVICE_UNAVAILABLE
        }.futureValue
````

Reply:
Hm, would it be better then to do something like this or might i just _.recoverWith(Left(ErrorDetails("fill in the blanks")))_?


def addFi(fiDetails: CreateFIDetails)(implicit
    hc: HeaderCarrier,
    ec: ExecutionContext
  ): Future[Either[**ServiceError**, HttpResponse]]

Reply:
current `upstreamError` which is being thrown - that may automatically gets logged into Elastic Search and kibana dashboard. If you silent it with `.recoverWith`, then will have to manually log